File: javascript/roman-numerals/roman-numerals.js
const symbolMapping = {
	I:1,
	V:5,
	X: 10,
	L: 50,
	C: 100,
	D: 500,
	M: 1000
};

export const toRoman = (number) => {
	if (number === 0) {
		return '';
	}
	else if (number < symbolMapping.V - symbolMapping.I) {
		return 'I'.repeat(number);
	}
	else if (number === symbolMapping.V - symbolMapping.I) {
		return 'IV'
	}
	else if (number < symbolMapping.X - symbolMapping.I) {
		return `V${'I'.repeat(number - symbolMapping.V)}`;
	}
	else if (number === symbolMapping.X - symbolMapping.I) {
		return 'IX';
	}
	else if (number < symbolMapping.L- symbolMapping.X) {
		return `X${toRoman(number - symbolMapping.X)}`;
	}
	else if (number < symbolMapping.L) {
		return `XL${toRoman(number - (symbolMapping.L - symbolMapping.X))}`;
	}
	else if (number < symbolMapping.C - symbolMapping.X) {
		return `L${toRoman(number - symbolMapping.L)}`;
	}
	else if (number < symbolMapping.C) {
		return `XC${toRoman(number - (symbolMapping.C - symbolMapping.X))}`;
	}
	else if (number < symbolMapping.D - symbolMapping.C) {
		return `C${toRoman(number - symbolMapping.C)}`;
	}
	else if (number < symbolMapping.D) {
		return `CD${toRoman(number - (symbolMapping.D - symbolMapping.C))}`;
	}
	else if (number < symbolMapping.M - symbolMapping.C) {
		return `D${toRoman(number - symbolMapping.D)}`;
	}
	else if (number < symbolMapping.M) {
		return `CM${toRoman(number - (symbolMapping.M - symbolMapping.C))}`;
	}
	else {
		return `M${toRoman(number - symbolMapping.M)}`;
	}
};


 

File: csharp/roman-numerals/RomanNumerals.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class RomanNumeralExtension
{
    public static string ToRoman(this int value)
    {
        List<RomanNumeralMapping> romanNumeralMapping = new List<RomanNumeralMapping>
        {
            new RomanNumeralMapping { Number = 1000, RomanNumeral = "M" },
            new RomanNumeralMapping { Number = 900,  RomanNumeral = "CM" },
            new RomanNumeralMapping { Number = 500,  RomanNumeral = "D" },
            new RomanNumeralMapping { Number = 400,  RomanNumeral = "CD" },
            new RomanNumeralMapping { Number = 100,  RomanNumeral = "C" },
            new RomanNumeralMapping { Number = 90,   RomanNumeral = "XC" },
            new RomanNumeralMapping { Number = 50,   RomanNumeral = "L" },
            new RomanNumeralMapping { Number = 40,   RomanNumeral = "XL" },
            new RomanNumeralMapping { Number = 10,   RomanNumeral = "X" },
            new RomanNumeralMapping { Number = 9,    RomanNumeral = "IX" },
            new RomanNumeralMapping { Number = 5,    RomanNumeral = "V" },
            new RomanNumeralMapping { Number = 4,    RomanNumeral = "IV" },
            new RomanNumeralMapping { Number = 1,    RomanNumeral = "I" }
        };

        var mappedSymbol = romanNumeralMapping.Find(x => x.Number <= value);
        if (mappedSymbol == null) { return string.Empty; }

        return mappedSymbol.Number == value ? mappedSymbol.RomanNumeral : $"{mappedSymbol.RomanNumeral}{(value - mappedSymbol.Number).ToRoman()}";
    }

    public class RomanNumeralMapping
    { 
        public int Number { get; set; }
        public string RomanNumeral { get; set; }
    }
}